libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -184,6 +183,10 @@ bool ReferencesResolver::visit(Return const& _return)
 
 void ReferencesResolver::operator()(yul::FunctionDefinition const& _function)
 {
+	checkYulIdentifierName(_function.name, _function.location);
+	for (yul::TypedName varName: _function.parameters + _function.returnVariables)
```

Comment:
This may be just a micro optimization, but maybe it's worth just passing by cref?

```suggestion
	for (yul::TypedName const& varName: _function.parameters + _function.returnVariables)
```